op-node/rollup/driver/state.go

```diff
@@ -278,11 +280,21 @@ func (s *Driver) eventLoop() {
 				return
 			}
 			if s.network != nil && payload != nil {
-				// Publishing of unsafe data via p2p is optional.
-				// Errors are not severe enough to change/halt sequencing but should be logged and metered.
-				if err := s.network.PublishL2Payload(s.driverCtx, payload); err != nil {
-					s.log.Warn("failed to publish newly created block", "id", payload.ID(), "err", err)
-					s.metrics.RecordPublishingError()
+				isActive, err := s.sequencerConductor.Active(s.driverCtx)
+				if err != nil {
+					s.log.Error("failed to determine if this node is the active sequencer, skipping publishing of unsafe data", "err", err)
+				} else if isActive {
+					// Publishing of unsafe data via p2p is optional.
+					// Errors are not severe enough to change/halt sequencing but should be logged and metered.
+					if err := s.network.PublishL2Payload(s.driverCtx, payload); err != nil {
+						s.log.Warn("failed to publish newly created block", "id", payload.ID(), "err", err)
+						s.metrics.RecordPublishingError()
+					}
+					if err := s.sequencerConductor.CommitUnsafePayload(s.driverCtx, payload); err != nil {
```

Comment:
We need to commit unsafe payload first, if and only if commit succeeded, can we publish payload to p2p.

This helps mitigate situations that sequencer failed around publishing payload to p2p, we have no way to guarantee if the payload reached the network or not, thus we'll have problem choosing which block to start on another sequencer.

With payload committed to consensus, we guarantee that payload is available to all sequencers (through p2p or conductor), and when starting sequencer, we'll always know which block to start from.

Reply:
Also if commit failed, we need to exit instead of continue to publish payload